src/v/archival/ntp_archiver_service.cc

```diff
@@ -2694,9 +2690,12 @@ ss::future<> ntp_archiver::garbage_collect() {
           "retry on the next housekeeping run.");
     }
 
-    _probe->segments_deleted(static_cast<int64_t>(successful_deletes));
+    _probe->segments_deleted(
+      static_cast<int64_t>(all_deletes_succeeded ? to_remove.size() : 0));
```

Comment:
Should we change the return type of `remote::delete_objects` to expose the count of keys that were not deleted? The S3 client returns the `undeleted_keys`, so we could use that.

Reply:
We can do that. The question is what we'll be doing with that value? We can't map it back to number of segments without doing some sort of mapping which I'd avoid as long as it is necessary to (to avoid introducing associative containers or iterating through results and mapping object keys back to segment names).

Change the semantics of segments deleted to objects deleted? Introduce a new metric for objects deleted?

Reply:
I was thinking we could update the probe on partial deletes to preserve the pre-existing behaviour.

Reply:
I understand that. But with what value? I'm don't have any idea how to get from number of objects that we failed to delete to number of segments without some sort of mapping.

The spillover gc isn't too great either. It behaves similarly but at a 1000 batch size granularity rather than unbound size .

Reply:
Ah I see what you're saying. I guess we could categorise the failures by object type within the s3 client, but I don't mind that much.